chore: update the glossary, faq and troubleshooting guide

Ngozi-Txfusion opened the pull request:
What

- put the faq, glossary and troubleshooting guide in the left bar
- use collapsible for the faq, glossary and troubleshooting guide
- remove the faq, glossary and troubleshooting guide from the main nav

Why

- Makes it easier for users to find the FAQ, Glossary, and Troubleshooting Guide.
- Reduces clutter in the main navigation.
- Collapsible sections provide a more organized and user-friendly layout.

Ngozi-Txfusion committed:
chore: update the glossary, faq and troubleshooting guide

What

- put the faq, glossary and troubleshooting guide in the left bar
- use collapsible for the faq, glossary and troubleshooting guide
- remove the faq, glossary and troubleshooting guide from the main nav

Why

- Makes it easier for users to find the FAQ, Glossary, and Troubleshooting Guide.
- Reduces clutter in the main navigation.
- Collapsible sections provide a more organized and user-friendly layout.

Ngozi-Txfusion committed:
chore: minor edit

itsacoyote commented on content/sdk/10.js/00.ethers/00.index.md:
```diff
@@ -68,7 +93,7 @@ to: https://join.zksync.dev
 title: Support Desk
 icon: i-heroicons-computer-desktop
-to: https://github.com/zksync-sdk/sdk-docs/blob/main/SUPPORT.md
+to: https://zksync.io/contact
```
Let's keep this pointing to the SUPPORT.md file. This file we have direct editing capabilities for. This also follows the GitHub open source community guidelines for project files. We don't have any access to edit that zksync.io/contact page.

Ngozi-Txfusion replied:
I have removed the entire page from this section

itsacoyote commented on content/sdk/10.js/00.ethers/00.index.md:
```diff
@@ -17,16 +17,41 @@ Take a look at the [migration guide](/sdk/js/ethers/v6/migration) if you are mig
 ### Installation
 
-To install SDK for zkSync Era, run the command below in your terminal.
+To install SDK for zkSync Era, follow the process below:
```
Could we clarify specifically which SDK they are installing in this page?

Ngozi-Txfusion replied:
I have added a callout to select either V5 and V6

itsacoyote commented on content/sdk/10.js/00.ethers/00.index.md:
```diff
@@ -17,16 +17,41 @@ Take a look at the [migration guide](/sdk/js/ethers/v6/migration) if you are mig
+
+    This will create a `package.json` file.
+
+#### Step 3: Install Required Packages
```
These number of steps seems unnecessary? They could have their project set up in a variety of ways and many might be looking to just integrate it into an existing project. We should just state that they install npm packages in a nodejs project at the most. The most important steps a user wants to see are just the install commands.

Ngozi-Txfusion replied:
Number steps are important for organizing documentation. 
Also, its helpful for newbies who wants to start using zksync-era except we are not targeting newbies.
I have re-arranged the steps.

itsacoyote replied:
I also agree that numbering is very useful for readability. What I meant was removing the steps on how to setup a project. As you said, we are not targeting newbies. They should know how to setup and manage their own projects, the primary purpose of this page is instructing them what packages to install. 
Also, the instructions you are providing for setting up a project are also primarily for linux systems, it may not work on windows systems.

If we want to demonstrate how to create a project and install packages, we can do so in our Community Code project.

Ngozi-Txfusion replied:
I have already removed the steps and updated it

itsacoyote reviewed (changes requested):
Just a few comments on the js/ethers page to have a look through

Ngozi-Txfusion committed:
chore: remove repetition as the js v5 and v6 are the same

- remove repetition as the js v5 and v6 are the same
- update the javascript v6 getting started page

Ngozi-Txfusion committed:
chore: minor update

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
```diff
@@ -1,69 +1,134 @@
 title: "Getting Started"
-description: "Explore zkSync-ethers SDK version 6"
-tags: ["zksync", "sdks", "zkSync Era SDK", "ethereum", "era network", "javascript", "v6"]
+description: "Explore zkSync-era SDK version 6"
```
Why is this renamed to zkSync-era? Isn't the package named zksync-ethers?

Ngozi-Txfusion replied:
Typo.. I have updated it

itsacoyote commented:
We should not remove any of the v5 pages from navigation. Just because v5 is obsolete does not mean that we can drop the documentation support for it. Adding the `_` to the files removes them from navigation and are no longer accessible.

Ngozi-Txfusion committed:
chore: add v5 in nav and update doc

Ngozi-Txfusion commented:
> We should not remove any of the v5 pages from navigation. Just because v5 is obsolete does not mean that we can drop the documentation support for it. Adding the `_` to the files removes them from navigation and are no longer accessible.

Since the content were repetition, that was why I removed them but I've added them now

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor link update

Ngozi-Txfusion committed:
chore: minor update

itsacoyote commented on content/sdk/10.js/00.ethers/00.index.md:
```diff
@@ -17,16 +17,19 @@ Take a look at the [migration guide](/sdk/js/ethers/v6/migration) if you are mig
-To install SDK for zkSync Era, run the command below in your terminal.
+To install SDK for zkSync Era, follow the process below:
+
+- Initialize a new Node.js project
```
Let's remove this line as well

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
```diff
@@ -1,69 +1,134 @@
+::
 
-## Examples
+- Create the scripts folder inside your project directory:
```
We need to re-work this section. It's too confusing with the blocks of code broken up. As I'm reading through the guide, I don't know where the code is in relation to the other code for the zksync.ts file. Take a look at other examples for Getting Started: 
- https://jestjs.io/docs/getting-started
- https://docs.ethers.org/v5/getting-started/
- https://viem.sh/docs/getting-started

Ngozi-Txfusion replied:
I tried including steps just like this: https://viem.sh/docs/getting-started
You specified that the steps shouldn't be added that's why I wrote it without the steps

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
````diff
@@ -1,69 +1,134 @@
+### Main Function
+
+```javascript
+async function main() {
````
The main function doesn't need to be described, it's just one way of writing and organizing this code. The closing bracket is missing as well. If this is to indicate the rest of the code is a part of the main function, we shouldn't break up the code like this. It makes it extremely hard to know where the rest of the code goes as someone is reading this guide.

Consider putting the code back together and using comments for your descriptions. For example like the Ethers.js doc's Getting Started code samples.

itsacoyote replied:
Yes, those blocks of code are broken down, but they are complete and runnable within their own context. The main function cannot run in the state that it is presented in this block.

Ngozi-Txfusion replied:
Okay, I understand..

I'll do that now

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
````diff
@@ -1,69 +1,134 @@
+```javascript
+  } catch (error) {
+    console.error(`Error: ${error}`);
+  }
````
This is another example of code that is too broken up. A catch can't stand on it's own, it needs to be related to a try but I don't know where this goes in relation to the code mentioned earlier.

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
````diff
@@ -1,69 +1,134 @@
-3. [Withdraw ETH and tokens from zkSync Era to Ethereum](https://github.com/zksync-sdk/zksync2-examples/blob/main/js/src/04_withdraw.ts)
-4. [Use paymaster to pay fee with token](https://github.com/zksync-sdk/zksync2-examples/blob/main/js/src/22_use_paymaster.ts)
+```shell
+npx ts-node src/zksync.ts
````
This doesn't look like the correct command to run, at the beginning you instructed to create the zksync.ts inside a /scripts folder.

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
```diff
@@ -1,69 +1,134 @@
-Full code for all examples is available [here](https://github.com/zksync-sdk/zksync2-examples/tree/main/js/src).
-Examples are configured to interact with `zkSync Era` and `Sepolia` test networks.
+::div{class="text-center"}
+![js sdk ethers](/images/sdk/js-ethers.png){ width="50%"}
```
We have a guideline to avoid using images for blocks of text. For logs like these, use the ```sh tag.

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -79,23 +73,18 @@ and paymaster input `0x8c5a3445`:
-[paymaster flows](https://docs.zksync.io/build/developer-reference/account-abstraction/paymasters#built-in-paymaster-flows)
+## Encoding Paymaster Parameters
+While the paymaster feature itself does not impose any limitations on the values of `paymasterInput`, the Matter Labs
+team endorses certain types of [paymaster flows](https://staging-docs.zksync.io/build/developer-reference/account-abstraction#built-in-paymaster-flows)
```
The URL here is incorrect, it's pointing to staging-docs. Use the production URL - docs.zksync.io

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -79,23 +73,18 @@ and paymaster input `0x8c5a3445`:
-[testnet paymaster](https://docs.zksync.io/build/developer-reference/account-abstraction/paymasters#testnet-paymaster):
+## See in Action
+Here’s how to call the method `setGreeting` of an `ethers` Contract object called `greeter`, while paying fees with the
+[testnet paymaster](https://staging-docs.zksync.io/build/developer-reference/account-abstraction#testnet-paymaster):
```
Remove `staging-`

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
````diff
@@ -121,3 +111,54 @@ const sentTx = await sender.sendTransaction({
+        paymasterParams,
+      },
+    });
+    ```
````
This is duplicating code which makes it harder to maintain. Can we move the explanations as comments inside the original code above?

itsacoyote commented on content/sdk/100.release-note.md:
```diff
@@ -2,7 +2,6 @@
 description: "Release Notes for SDKs"
 tags: ["zksync", "sdks", "zkSync Era SDK", "ethereum", "era network", "release notes"]
-navigation: false
 ---
```
Is there a reason this was removed? There is no content yet here, and I don't want any empty pages for release.

itsacoyote commented on content/sdk/80.troubleshooting.md:
```diff
@@ -8,61 +8,67 @@ navigation: false
+
+1. **Network Selection**: Ensure you are using the Layer 2 (L2) network instead of Layer 1 (L1) Sepolia. The correct
+endpoint for the
+L2 network is: [zkSync Sepolia Explorer](https://sepolia.explorer.zksync.io/).
```
We have tags that make referring to common values like the zkSync sepolia explorer easier to manage. Replace the URL with `%%zk_testnet_block_explorer_url%%`

itsacoyote commented on content/sdk/80.troubleshooting.md:
```diff
@@ -8,61 +8,67 @@ navigation: false
+## JavaScript SDK
+
+  ::collapsible
+    :field If you encounter issues using Js SDK during the installation, refer to the troubleshooting steps:
```
This is not the correct format for the `:field` component. 
`:field{name=""}`
But this also doesn't seem to be an appropriate use of the field component for this bit of text.

itsacoyote replied:
The issue here is that `:field` you're using isn't valid so it's not even formatting anything correctly with that.

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor edit

Ngozi-Txfusion committed:
chore: minor edit

Ngozi-Txfusion committed:
chore: minor changes

Ngozi-Txfusion committed:
chore: minor update

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
```diff
@@ -1,69 +1,154 @@
 
-To begin, it is useful to have a basic understanding of the types of objects available and what they are
-responsible for, at a high level:
+This guide will help you interact with the zkSync Era blockchain using the zkSync JavaScript SDK. We'll walk through
```
Let's clarify this is the zksync-ethers SDK, not the javascript sdk. There may be more javascript libraries we will support later on.

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
```diff
@@ -1,69 +1,154 @@
+
+- Node.js
+- TypeScript
+- `zksync-ethers` library
```
I know we have the Getting Started guide at the Ethers section, but we need the Getting Started guide at this level explain how the install the necessary library. Remove the prerequisites and replace it with just instructions on what packages to install for zksync-ethers.

Ngozi-Txfusion replied:
ok

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
````diff
@@ -1,69 +1,154 @@
-const ethProvider = ethers.getDefaultProvider("sepolia");
+```bash
+touch scripts/zksync.ts
+```
````
Can we remove the first step? Is there a reason to create a scripts folder? That adds an extra step that is unnecessary but makes it feel like it's a requirement to use this library. We don't know what their project setup looks like and shouldn't force them to take additional unnecessary steps.

Ngozi-Txfusion replied:
I've added an intro

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -79,25 +71,29 @@ and paymaster input `0x8c5a3445`:
 
-zkSync SDK provides a utility method that can be used to get the correctly formed `paymasterParams`
-object: [getPaymasterParams](/sdk/js/ethers/v6/paymaster-utils#getpaymasterparams).
+The zkSync SDK includes a utility method for forming `paymasterParams` correctly, which is especially useful for
```
```suggestion
The zksync-ethers SDK includes a utility method for forming `paymasterParams` correctly, which is especially useful for
```

itsacoyote commented on pages/sdk/[...slug].vue:
```diff
@@ -44,7 +44,7 @@ const links = computed(() =>
     {
       icon: 'i-heroicons-pencil-square',
-      label: 'Edit this page',
+      label: 'Suggest changes to this page',
```
Let's keep this to "Edit this page" to keep the length short.

itsacoyote commented on content/sdk/10.js/00.ethers/00.index.md:
```diff
@@ -51,24 +53,3 @@ to: /sdk/js/ethers/v6/getting-started
 Integrate zksync GO SDK for zkSync Era features.
 ::
 ::
```
These `::` should be removed.

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
```diff
@@ -1,69 +1,154 @@
 title: "Getting Started"
 description: "Explore zkSync-ethers SDK version 6"
-tags: ["zksync", "sdks", "zkSync Era SDK", "ethereum", "era network", "javascript", "v6"]
+tags: ["zksync", "sdks", "zkSync Era SDK", "ethereum", "era network", "javascript", v6"]
```
```suggestion
tags: ["zksync", "sdks", "zkSync Era SDK", "ethereum", "era network", "javascript", "v6"]
```

itsacoyote reviewed (changes requested):
It's gotten great progress so far, just a few more things to suggest changing 👍🏻

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/01.features.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/00.getting-started.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
Merge branch 'update-sdk-doc' of https://github.com/zksync-sdk/sdk-docs into update-sdk-doc

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor changes

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor changes

Ngozi-Txfusion committed:
feat: updated the look and feel

- added the sdk selector
- added accordion to the faq
- updated the look of the homepage
- added a compatibility page

Ngozi-Txfusion committed:
Merge branch 'main' into update-sdk-doc

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor edit

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor uodate

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -1,59 +1,50 @@
 
-While ZKsync is mostly `ethers`-compatible, it has some differences compared to Ethereum.
-The major of those are:
+zksync Era offers several unique features compared to Ethereum, which require some specific handling. This guide will
```
```suggestion
ZKsync Era offers several unique features compared to Ethereum, which require some specific handling. This guide will
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -1,59 +1,50 @@
-[here](https://docs.zksync.io/zk-stack/concepts/transaction-lifecycle.html#eip-712-0x71).
+::card-group
+::card{title="Account Abstraction" icon="i-heroicons-puzzle-piece"}
+zksync supports account abstraction, allowing accounts to have custom validation logic and enabling paymaster support.
```
```suggestion
ZKsync Era supports account abstraction, allowing accounts to have custom validation logic and enabling paymaster support.
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -1,59 +1,50 @@
+zksync supports account abstraction, allowing accounts to have custom validation logic and enabling paymaster support.
+::
+::card{title="Deployment Transactions" icon="i-heroicons-paper-airplane"}
+Contracts on zksync require bytecode to be passed in a separate field during deployment.
```
```suggestion
Contracts on ZKsync Era require bytecode to be passed in a separate field during deployment.
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -1,59 +1,50 @@
 ---
-title: ZKsync Era Features
-description:
+title: "zksync Era Features"
```
```suggestion
title: "ZKsync Era Features"
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -1,59 +1,50 @@
+::card{title="Fee System" icon="i-heroicons-banknotes"}
+The fee system in zksync differs from Ethereum's, necessitating additional fields in transactions.
+::
+::
```
Please change this to a bulleted list. The way we currently use cards as links to a page make these look like you can click on them.

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -1,59 +1,50 @@
+## EIP712 Transactions
 
-## Overrides
+To accommodate these differences, zksync extends standard Ethereum transactions with new fields, creating what are known
```
```suggestion
To accommodate these differences, ZKsync Era extends standard Ethereum transactions with new fields, creating what are known
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -1,59 +1,50 @@
 
-## Overrides
+To accommodate these differences, zksync extends standard Ethereum transactions with new fields, creating what are known
+as EIP712 transactions. More details on EIP712 transactions can be found [here](https://docs.zksync.io/zk-stack/concepts/transaction-lifecycle.html#eip-712-0x71).
```
```suggestion
as EIP712 transactions. More details on EIP712 transactions can be found [here](https://docs.zksync.io/zk-stack/concepts/transaction-lifecycle#eip-712-0x71).
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -1,59 +1,50 @@
-In this case, you can provide an `Overrides` object as the last parameter.
-There you can supply fields like `gasPrice`, `gasLimit`, `nonce` etc.
+`ethers.js` provides overrides for transactions, allowing you to set values like `gasPrice`, `gasLimit`, and `nonce`.
+zksync-specific fields are added using the `customData` object in these overrides.
```
This feels misdirecting with the wording. Yes it is added using `customData` to set these properties, but you removed the example and description that the `customData` has to be a property of the `overrides` key.

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -79,25 +69,29 @@ and paymaster input `0x8c5a3445`:
-## Encoding paymaster params
+::callout{icon="i-heroicons-light-bulb"}
+Everything inside `customData` in `overrides` is related to zksync(L2 gas, etc.).
+::
```
This should be a part of the description up at the beginning. This is too far down the page to mention.

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/02.front-end.md:
```diff
@@ -1,16 +1,17 @@
 ---
 
-This section describes how to make the most of ZKsync to provide the best UX.
+Learn how to integrate zksync into your front-end for the best user experience.
```
```suggestion
Learn how to integrate ZKsync Era into your front-end for the best user experience.
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/02.front-end.md:
```diff
@@ -1,16 +1,17 @@
-title: Front-end Integration
-description:
+title: Frontend Integration
+description: "How to integrate zksync for optimal user experience"
```
```suggestion
description: "How to integrate ZKsync Era for optimal user experience"
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/02.front-end.md:
```diff
@@ -1,16 +1,17 @@
+infrastructure will work seamlessly.
 
-## Enabling ZKsync features
+## Using zksync Features
```
```suggestion
## Using ZKsync Era Features
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/02.front-end.md:
```diff
@@ -1,16 +1,17 @@
 
-If you want to deploy smart contracts or enable advanced ZKsync features, like account abstraction,
-then you need to use the `zksync-ethers` library. You can read about the basics [here](/sdk/js/ethers/v6/features).
+To deploy smart contracts or use advanced zksync features like account abstraction, you'll need the `zksync-ethers`
```
```suggestion
To deploy smart contracts or use advanced ZKsync Era features like account abstraction, you'll need the `zksync-ethers`
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/02.front-end.md:
```diff
@@ -1,16 +1,17 @@
-If you want to deploy smart contracts or enable advanced ZKsync features, like account abstraction,
-then you need to use the `zksync-ethers` library. You can read about the basics [here](/sdk/js/ethers/v6/features).
+To deploy smart contracts or use advanced zksync features like account abstraction, you'll need the `zksync-ethers`
+library. Learn the basics [here](/sdk/js/ethers/v6/features).
```
Not a requirement for this PR, but let's try to improve our sentences with links to better describe the linked page: https://developers.google.com/style/link-text

itsacoyote commented on pages/sdk/[...slug].vue:
```diff
@@ -73,6 +90,19 @@ const links = computed(() =>
+        class="mt-10"
+      >
+        <GlossaryAccordion />
+      </p>
```
While the accordion does make things look succinct on the page, can we please remove them and just use a list of H2 headers? The content gets hidden away and the Table of Contents is empty, this makes it bad for accessibility.

itsacoyote reviewed (changes requested):
I've added some comments to adjust some things on the pages. Also please double check that you are using `ZKsync` instead of `zkSync`, the branding has been updated.

itsacoyote commented on .DS_Store:
This should not be added to the project, please delete.

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/02.front-end.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/02.front-end.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/02.front-end.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
Merge branch 'update-sdk-doc' of https://github.com/zksync-sdk/sdk-docs into update-sdk-doc

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/00.index.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/00.index.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/00.getting-started.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/02.front-end.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/00.getting-started.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/01.features.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/10.v5/00.getting-started.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
chore: minor changes

Ngozi-Txfusion committed:
Merge branch 'update-sdk-doc' of https://github.com/zksync-sdk/sdk-docs into update-sdk-doc

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/10.v5/00.getting-started.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/00.index.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
Merge branch 'update-sdk-doc' of https://github.com/zksync-sdk/sdk-docs into update-sdk-doc

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -1,35 +1,63 @@
-To accommodate these differences, zksync extends standard Ethereum transactions with new fields, creating what are known
-as EIP712 transactions. More details on EIP712 transactions can be found [here](https://docs.zksync.io/zk-stack/concepts/transaction-lifecycle.html#eip-712-0x71).
+To accommodate these differences, ZKsync extends standard Ethereum transactions with new fields, creating what are known
+as EIP712 transactions. More details on EIP712 transactions can be found in the [ZKstack transaction lifecycle](https://docs.zksync.io/zk-stack/concepts/transaction-lifecycle.html#eip-712-0x71).
```
The URL is not the correct url.

```suggestion
as EIP712 transactions. More details on EIP712 transactions can be found in the [ZKstack transaction lifecycle](https://docs.zksync.io/zk-stack/concepts/transaction-lifecycle#eip-712-0x71).
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/02.front-end.md:
```diff
@@ -1,17 +1,17 @@
 infrastructure will work seamlessly.
 
-## Using zksync Features
+## Using zksync era features
```
When they're proper nouns you can use the proper capitalization in headers. 

```suggestion
## Using ZKsync Era features
```

itsacoyote commented on content/sdk/00.index.md:
```diff
@@ -1,14 +1,31 @@
+
+::div{class="text-center"}
+![ZKsync Era SDKs](/images/sdk/zksync-era-sdks.png){ width="50%"}
+::
```
We need to remove this image for now. It is not using the latest branding design.

itsacoyote commented on content/sdk/00.index.md:
```diff
@@ -1,14 +1,31 @@
+**ZKsync Era** is a Layer 2 solution for Ethereum. It uses a special technology called **ZK rollup**. This technology
+makes transactions faster and cheaper while keeping the security and trust of the Ethereum blockchain.
+
+### Key Points
```
```suggestion
### Key points
```

itsacoyote commented on content/sdk/10.js/00.ethers/00.index.md:
```diff
@@ -1,14 +1,14 @@
 ---
-title: "Getting Started"
+title: "Overview"
 description: "Explore ZKsync-ethers SDK versions"
```
```suggestion
description: "Explore zksync-ethers SDK versions"
```

itsacoyote commented on content/sdk/10.js/00.ethers/00.index.md:
```diff
@@ -1,14 +1,14 @@
 
 ::callout
-Take a look at the [migration guide](/sdk/js/ethers/v6/migration) if you are migrating from `zksync-web3`.
+Take a look at the [migration guide](/sdk/js/ethers/v6/migration) if you are migrating from `ZKsync-web3`.
```
```suggestion
Take a look at the [migration guide](/sdk/js/ethers/v6/migration) if you are migrating from `zksync-web3`.
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
````diff
@@ -1,69 +1,130 @@
-```shell
-npx ts-node src/zksync.ts
+```bash
+npm install --save-dev typescript
````
Let's remove the typescript install part. All they should need is the `npm install zksync-ethers ethers` step. 
Typescript can be installed globally or locally and depending on the user's setup you could cause issues making them install it locally. It is also already defined as a devDependency for the package.

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
````diff
@@ -1,69 +1,130 @@
 ```
 
-Get the network (helper function [toJSON](/sdk/js/ethers/v6/providers#tojson)):
+## Network Information
````
```suggestion
## Network information
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
````diff
@@ -1,69 +1,130 @@
 ```
 
-Get the latest block number:
+## Transaction Details
````
```suggestion
## Transaction details
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
````diff
@@ -1,69 +1,130 @@
 ```
 
-Get the block by hash (helper function [toJSON](/sdk/js/ethers/v6/providers#tojson)):
+## Error Handling
````
```suggestion
## Error handling
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
````diff
@@ -1,69 +1,130 @@
 ```
 
-Get the transaction by hash (helper function [toJSON](/sdk/js/ethers/v6/providers#tojson)):
+## Example Code
````
```suggestion
## Example code
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
````diff
@@ -1,69 +1,130 @@
+  .catch((error) => console.error(`Error executing script: ${error}`));
+```
+
+## Run the Script
````
```suggestion
## Run the script
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -1,59 +1,78 @@
-title: ZKsync Era Features
-description:
+title: "ZKsync Era Features"
+description: "Explore ZKsync Era Features and Custom Transactions"
```
```suggestion
description: "Explore ZKsync Era features and custom transactions"
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/01.features.md:
```diff
@@ -1,59 +1,78 @@
+::
 
-## Overrides
+## EIP712 Transactions
```
```suggestion
## EIP712 transactions
```

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

itsacoyote commented on content/sdk/80.troubleshooting.md:
```diff
@@ -1,68 +1,66 @@
+- **Possible Cause**: Conflicts with existing aliases or functions
+- **Solution**: Ensure there are no conflicting aliases or functions named `bun` by running: `type bun`
+
+##### 8. Reinitialize Shell
```
Update these headers to follow the documentation styleguide for headers.

itsacoyote commented on content/sdk/90.glossary.md:
```diff
@@ -1,8 +1,68 @@
+An environment variable in Unix-like operating systems and Windows, specifying a set of directories where executable
+programs are located.
+
+### Shell Configuration File
```
```suggestion
### Shell configuration file
```

itsacoyote reviewed (changes requested):
It's looking great now! Just a few more adjustments for styleguide rules and it'll be ready to merge 👍🏻

Ngozi-Txfusion committed:
Update content/sdk/90.glossary.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/00.index.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/01.features.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/01.features.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/00.index.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/00.index.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/00.getting-started.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/00.getting-started.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/00.getting-started.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/02.front-end.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/01.features.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/00.getting-started.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/00.getting-started.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
chore: minor update

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/02.front-end.md:
```diff
@@ -1,16 +1,17 @@
-title: Front-end Integration
-description:
+title: Frontend Integration
+description: "How to integrate ZKsync Era for optimal user experience"
```
```suggestion
description: How to integrate ZKsync Era for optimal user experience
```

itsacoyote commented on content/sdk/10.js/00.ethers/20.v6/00.getting-started.md:
```diff
@@ -1,69 +1,121 @@
+## Initialization
 
-The following examples could be easily run by writing the code snippets in a file and executing them as follows:
+To initialize the provider to connect to the ZKsync Sepolia network:
```
It looks like the install zksync-ethers step is missing? `npm install zksync-ethers ethers` Only the typescript install should be removed.

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
Update content/sdk/10.js/00.ethers/20.v6/02.front-end.md

Co-authored-by: Sabrina <sf@matterlabs.dev>

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

Ngozi-Txfusion committed:
chore: minor update

itsacoyote reviewed (commented):
🎉
